api/revisions/serializers.py

```diff
@@ -0,0 +1,155 @@
+from rest_framework import exceptions
+
+from api.base.serializers import (
+    RelationshipField,
+    VersionedDateTimeField,
+)
+
+from osf.models import (
+    Registration,
+    SchemaResponses,
+    SchemaResponseBlock,
+    RegistrationSchemaBlock,
+)
+
+
+class SchemaResponsesSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'date_created',
+        'date_modified',
+        'revision_justification',
+        'reviews_state',
+    ])
+
+    id = ser.CharField(source='_id', required=False, allow_null=True)
+    date_created = VersionedDateTimeField(source='created', required=False)
+    date_modified = VersionedDateTimeField(source='modified', required=False)
+    revision_justification = ser.CharField(required=False)
+    revision_response = ser.JSONField(source='schema_responses', required=False)
+    reviews_state = ser.ChoiceField(choices=['revision_in_progress', 'revision_pending_admin_approval', 'revision_pending_moderation', 'approved'], required=False)
+    is_pending_current_user_approval = ser.SerializerMethodField()
+
+    links = LinksField(
+        {
+            'self': 'get_absolute_url',
+        },
+    )
+
+    registration = RelationshipField(
+        related_view='registrations:registration-detail',
+        related_view_kwargs={'node_id': '<parent._id>'},
+        required=False,
+    )
+
+    registration_schema = RelationshipField(
+        related_view='schemas:registration-schema-detail',
+        related_view_kwargs={'schema_id': '<parent.registered_schema_id>'},
+        read_only=True,
+        required=False,
+    )
+
+    initiated_by = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<initiator._id>'},
+        read_only=True,
+        required=False,
+
+    )
+
+    class Meta:
+        type_ = 'revisions'
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse(
+            'revisions:schema-responses-detail',
+            kwargs={
+                'version': self.context['request'].parser_context['kwargs']['version'],
+                'revision_id': obj._id,
+            },
+        )
+
+    def get_is_pending_current_user_approval(self, obj):
+        # TBD
+        return False
+
+
+class SchemaResponsesListSerializer(SchemaResponsesSerializer):
+
+    def create(self, validated_data):
+        registration = Registration.load(validated_data.pop('_id'))
+        if registration.registered_schema.first():
+            schema_response = SchemaResponses.objects.create(
+                **validated_data,
+                object_id=registration.id,
+                content_type=ContentType.objects.get_for_model(registration),
+                initiator=self.context['request'].user
+            )
+        else:
+            raise NotImplementedError()
+
+        return schema_response
+
+
+class SchemaResponsesDetailSerializer(SchemaResponsesSerializer):
+
+    versions = RelationshipField(
```

Comment:
Not seeing where this is called for. Don't think we need it.

Reply:
removed